kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerOffsetsDaoImpl.java

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import io.confluent.kafkarest.config.ConfigModule.OffsetsTimeoutConfig;
+import io.confluent.kafkarest.entities.ConsumerGroupLag;
+import io.confluent.kafkarest.entities.ConsumerLag;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.ConsumerGroupListing;
+import org.apache.kafka.clients.admin.DescribeConsumerGroupsOptions;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListConsumerGroupsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsResult;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.admin.MemberDescription;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+
+final class ConsumerOffsetsDaoImpl implements ConsumerOffsetsDao {
+
+  private final Admin kafkaAdminClient;
+  private final int consumerMetadataTimeout;
+
+  @Inject
+  ConsumerOffsetsDaoImpl(
+      Admin kafkaAdminClient,
+      @OffsetsTimeoutConfig Duration consumerMetadataTimeout
+  ) {
+    this.kafkaAdminClient = kafkaAdminClient;
+    // ahu todo: figure out best way to safely cast to int (required by AbstractOptions.timeoutMs())
+    // this.consumerMetadataTimeout = Math.toIntExact(Long.MAX_VALUE);
+    this.consumerMetadataTimeout = Math.toIntExact(
+        Math.min(consumerMetadataTimeout.toMillis(), Integer.MAX_VALUE));
```

Comment:
capped at Integer.MAX_VALUE to avoid ArithmeticException
(OFFSETS_TIMEOUT_DOC in KafkaRestConfig also notes Integer.MAX_VALUE is the largest value)